File: MsgTools/struct/lua/DecodeStructmap.lua
	    
		local @paramName@Temp = {}
		local @paramName@size = inputStream:ReadInt()
		for i=1,@paramName@size do
		  local key = inputStream:ReadUTFString()
          @valueMapDecode@
		end
		body.@paramName@ = @paramName@Temp

File: MsgTools/struct/java/DecodeStructmap.java
        
        int @paramName@Size = inputStream.readInt();
		if(@paramName@Size>0){
			@paramName@ = new HashMap<String,@typeName@>();
			for(int @paramName@Cursor=0;@paramName@Cursor<@paramName@Size;@paramName@Cursor++){
				String key = inputStream.readUTF();
				@valueMapDecode@
			}
		}